src/bio/bio_xstream.c

```diff
@@ -908,10 +934,50 @@ init_bio_bdevs(struct bio_xs_context *ctxt)
 		if (nvme_glb.bd_bdev_class != get_bdev_type(bdev))
 			continue;
 
-		rc = create_bio_bdev(ctxt, spdk_bdev_get_name(bdev), NULL);
+		rc = bdev_name2roles(spdk_bdev_get_name(bdev));
+		if (rc < 0)
+			break;
+
+		/* skip non metadata devices */
+		if (!(rc & NVME_ROLE_INDEX))
+			continue;
+
+		rc = create_bio_bdev(ctxt, spdk_bdev_get_name(bdev), NULL, true);
 		if (rc)
 			break;
+
+		*sys_bdev = bdev;
+		break;
 	}
+
+	return rc;
+}
+
+static int
+init_bio_nosys_bdevs(struct bio_xs_context *ctxt, struct spdk_bdev *except_bdev)
```

Comment:
I don't see why do we need separate init_bio_sys_bdev() and init_bio_nosys_bdevs().